controllers/userStatus.js

```diff
@@ -56,6 +56,45 @@ const getUserStatus = async (req, res) => {
   }
 };
 
+/**
+ * Returns sorted user status on the basis of how long they have had that status for and updates user's from field
+ *
+ * @param req {Object} - All user object
+ */
+
+const getSortedUsers = (allUsers) => {
+  const sorted = allUsers.sort((a, b) => a.currentStatus.from - b.currentStatus.from);
+  const today = new Date();
+  sorted.forEach((data) => {
+    const idleSincedate = new Date(data.currentStatus.from);
+    const dateArray = idleSincedate.toISOString().split("-");
+    const year = today.getFullYear();
+    const month = today.getMonth() + 1;
+    const day = today.getDate();
+    const yy = parseInt(dateArray[0]);
+    const mm = parseInt(dateArray[1]);
+    const dd = parseInt(dateArray[2]);
+    let years, months;
+    // months
+    months = month - mm;
+    if (day < dd) {
+      months = months - 1;
+    }
+    // years
+    years = year - yy;
+    if (month * 100 + day < mm * 100 + dd) {
+      years = years - 1;
+      months = months + 12;
+    }
+    // days
+    const days = Math.floor(
+      (today.getTime() - new Date(yy + years, mm + months - 1, dd).getTime()) / (24 * 60 * 60 * 1000)
+    );
+    //
+    data.currentStatus.from = `${years} years , ${months} months , ${days} days`;
```

Comment:
Because I want to use it in the frontend and there, I want to show the number of days, month and year the user has been idle for. Thought it will be a good idea to handle it on backend.

Reply:
can we think of adding a new field to achieve this? what if the Frontend is also interested to know the timestamp as well. Overwriting `from` will erase the timestamp data.

Reply:
Then, on frontend, they can simply remove the query  `sorted` to show the timestamp? or should it be handled better?

Reply:
what if the requester is interested in knowing the timestamp as well as the number of days he has been idle since? The best I could think of is instead of modifying the `from` field we can give a new key to convey the no of days idle since information but I am not sure what's the best approach. Maybe you can check with @prakashchoudhary07 and decide the way forward.